src/app/modules/myinfo/myinfo.util.ts

```diff
@@ -285,16 +286,23 @@ export const createRelayState = (formId: string): string =>
  * Validates that a form is a MyInfo form with an e-service ID
  * @param form Form to validate
  */
-export const validateMyInfoForm = (
-  form: IFormSchema | IPopulatedForm,
-): Result<IMyInfoForm, MyInfoNoESrvcIdError | MyInfoAuthTypeError> => {
+export const validateMyInfoForm = <T extends IFormSchema>(
+  form: T,
+): Result<MyInfoForm<T>, FormAuthNoEsrvcIdError | AuthTypeMismatchError> => {
   if (!form.esrvcId) {
-    return err(new MyInfoNoESrvcIdError())
+    return err(new FormAuthNoEsrvcIdError(form._id))
   }
-  if (form.authType !== AuthType.MyInfo) {
-    return err(new MyInfoAuthTypeError())
+  if (isMyInfoForm(form)) {
+    return ok(form)
   }
-  return ok(form as IMyInfoForm)
+  return err(new AuthTypeMismatchError(AuthType.MyInfo, form.authType))
+}
+
+// Typeguard to ensure that form has eserviceId and MyInfo authType
+const isMyInfoForm = <F extends IFormSchema>(
```

Comment:
technically a MyInfo form can exist without an e-service ID as long as it's not public. can we name this more specifically to reflect that? eg isMyInfoFormWithEsrvcId

Reply:
oh yea, definitely. good catch and i'll do so!